File: ios/HealthInfo.m
//
//  HealthInfo.m
//  HealthApp
//
//  Created by Just1and0 on 01/03/2024.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(HealthInfo, NSObject)

RCT_EXTERN_METHOD(authorizeHealthKit:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(getHeartRate:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(getBodyTemperature:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(getOxygenSaturation:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(getBloodPressureSystolic:(RCTResponseSenderBlock)callback)
@end

File: android/app/src/main/java/com/healthapp/HealthInfo.java
package com.healthapp;

import androidx.annotation.NonNull;
import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class HealthInfo extends ReactContextBaseJavaModule {
    public HealthInfo(ReactApplicationContext context) {
        super(context);
    }

    @ReactMethod
    public void authorizeHealthKit(Callback callback) {
        callback.invoke(false);
    }

    @ReactMethod
    public void getHeartRate(Callback callback){
        callback.invoke(false);
    }

    @ReactMethod
    public void getBodyTemperature(Callback callback){
        callback.invoke(false);
    }

    @ReactMethod
    public void getOxygenSaturation(Callback callback){
        callback.invoke(false);
    }

    @ReactMethod
    public void getBloodPressureSystolic(Callback callback){
        callback.invoke(false);
    }

    @NonNull
    @Override
    public String getName() {
        return "HealthInfo";
    }

}


